File: app/src/main/java/com/hugocastelani/waterfalltoolbar/sample/ItemModel.java
package com.hugocastelani.waterfalltoolbar.sample;

import android.support.annotation.NonNull;

/**
 * Created by Hugo Castelani
 * Date: 20/09/17
 * Time: 17:38
 */

public class ItemModel {
    private String title;
    private String summary;

    public ItemModel(@NonNull final String title, @NonNull final String summary) {
        this.title = title;
        this.summary = summary;
    }

    @NonNull
    public String getTitle() {
        return title;
    }

    public void setTitle(@NonNull final String title) {
        this.title = title;
    }

    @NonNull
    public String getSummary() {
        return summary;
    }

    public void setSummary(@NonNull final String summary) {
        this.summary = summary;
    }

    @Override
    public String toString() {
        return title + "\n" + summary;
    }
}

File: app/src/main/java/com/hugocastelani/waterfalltoolbar/sample/sample/ItemModel.kt
package com.hugocastelani.waterfalltoolbar.sample.sample

/**
 * Created by Hugo Castelani
 * Date: 20/09/17
 * Time: 17:38
 */

class ItemModel() {
    var title: String? = null
    var summary: String? = null

    constructor(title: String?, summary: String?): this() {
        this.title = title
        this.summary = summary
    }

    override fun toString(): String = "$title\n$summary"
}

fun itemModel(init: ItemModel.() -> Unit): ItemModel {
    val itemModel = ItemModel()
    itemModel.init()
    return itemModel
}